woocommerce/woocommerce
# Fix: Enable Pagination Block Visibility in Product Collection Block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR resolves an issue where the Pagination block was not visible in the block insertor within the Product Collection block.

- The core query pagination block name, initially hardcoded, is now declared as a constant `coreQueryPaginationBlockName` in `constants.ts`. This change not only improves code readability but also ensures consistency in using the block name across different files.

- In `index.tsx`, the `addProductCollectionBlockToParentOfPaginationBlock` utility function from `utils.tsx` is imported and executed. This function plays a crucial role in the fix. It dynamically adds the Product Collection block to the parent array of the Core Pagination block, ensuring the latter's visibility in the insertor when editing the Product Collection block.

With these changes, merchants can now easily add a Pagination block from the insertor while working on the Product Collection block, enhancing the user experience and functionality.

Closes #42194 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new post
2. Add Product Collection block to it & choose “Product Catalog” 
3. There is already one pagination block. Add another pagination block to Product Collection block as shown below:

<img src="https://github.com/woocommerce/woocommerce/assets/16707866/d60cc099-205c-47c4-a649-3e65d03344e1" width="300">


4. Verify that:
    1. You can add another pagination block using insertor
    2. After publishing, check on the frontend to ensure both pagination blocks are functioning correctly.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Fix: Pagination Block Visibility in Product Collection Block

#### Comment 

</details>
